django/db/models/query.py

```diff
@@ -454,6 +454,8 @@ def bulk_create(self, objs, batch_size=None):
                         assert len(ids) == len(objs_without_pk)
                     for i in range(len(ids)):
                         objs_without_pk[i].pk = ids[i]
+                        objs_without_pk[i]._state.adding = False
+                        objs_without_pk[i]._state.db = self.db
```

Comment:
That reads so much more Pythonic, agreed. The caching of `self.db` seems overkill, but the rest makes a lot of sense.